xcom2-launcher/xcom2-launcher/Forms/MainForm.Events.cs

```diff
@@ -65,14 +66,18 @@ internal void RegisterEvents()
             // Edit
             editOptionsToolStripMenuItem.Click += delegate
             {
-                var result = new SettingsDialog(Settings).ShowDialog();
+                var dialog = new SettingsDialog(Settings);
 
-                if (result == DialogResult.OK)
+                if (dialog.ShowDialog() == DialogResult.OK)
                 {
+                    // refresh/update settings dependent functions
                     RefreshModList();
-                    ShowQuickLaunchArgsBasedOnSettings();
                     showHiddenModsToolStripMenuItem.Checked = Settings.ShowHiddenElements;
                     UpdateQuickArgumentsMenu();
+
+                    if (dialog.IsRestartRequired) {
+                        MessageBox.Show("Some changes won't take effect, until after the application has been restarted.", "Info", MessageBoxButtons.OK, MessageBoxIcon.Information);
```

Comment:
Can't we leave a more persistent (but small) notification somewhere?

Reply:
Maybe show a label saying "Application restart pending" (probably with an additional tooltip) on the main form?

Reply:
👍